core/src/wallet/common/api_impl/OperationApi.cpp

```diff
@@ -135,6 +138,13 @@ namespace ledger {
             return std::make_shared<EthereumLikeOperation>(shared_from_this());
         }
 
+        std::shared_ptr<api::RippleLikeOperation> OperationApi::asRippleLikeOperation() {
+            if (getWalletType() != api::WalletType::RIPPLE) {
```

Comment:
This is the kind of thing the new architecture must not have: if we want to downcast (and I hope we’ll settle on another design), the fact the downcast can fail should be conveyed in the API (with a `Result` / `Option` type).

Reply:
Yes it would be better having a `Result` type